artifactory/commands/transferfiles/timeestimation_test.go

```diff
@@ -0,0 +1,227 @@
+		Files: []FileUploadStatusResponse{
+			{
+				Status:           Success,
+				ChecksumDeployed: true,
+				SizeBytes:        8 * bytesInMB,
+			},
+		},
+	}
+	timeEstMng.addChunkStatus(chunkStatus1, 3, true)
+	assert.Equal(t, "Not available yet", timeEstMng.getSpeedString())
+	assert.Equal(t, "Not available yet", timeEstMng.getEstimatedRemainingTimeString())
+}
+
+func newDefaultTimeEstimationManager() *timeEstimationManager {
+	return newTimeEstimationManager(600*bytesInMB, 100*bytesInMB)
+}
+
+func TestAddingToFullLastSpeedsSlice(t *testing.T) {
+	timeEstMng := newDefaultTimeEstimationManager()
+	numOfSpeedsToKeepPerWorkingThread = 3
+
+	// Fill the last speeds slice (up to threads * numOfSpeedsToKeepPerWorkingThread).
+	timeEstMng.lastSpeeds = []float64{1.1, 2.2, 3.3, 4.4, 5.5, 6.6}
+
+	// Add a chunk and assert the oldest speed is removed, new is added, and the chunk len remains the same.
+	firstChunkSpeed := addOneFileChunk(timeEstMng, 2, 20, 1)
+	assert.Equal(t, []float64{2.2, 3.3, 4.4, 5.5, 6.6, firstChunkSpeed}, timeEstMng.lastSpeeds)
+
+	// Lower threads and add a chunk. Expecting the slice to shrink and the new speed to be added in the end.
+	secondChunkSpeed := addOneFileChunk(timeEstMng, 1, 30, 2)
+	assert.Equal(t, []float64{6.6, firstChunkSpeed, secondChunkSpeed}, timeEstMng.lastSpeeds)
+
+	// Increase threads and add a chunk. Expecting the slice len to increase with the new speed.
+	thirdChunkSpeed := addOneFileChunk(timeEstMng, 3, 40, 3)
+	assert.Equal(t, []float64{6.6, firstChunkSpeed, secondChunkSpeed, thirdChunkSpeed}, timeEstMng.lastSpeeds)
+}
+
+// Adds a chunk with one non checksum-deployed file and calculates and returns the chunk speed.
+func addOneFileChunk(timeEstMng *timeEstimationManager, workingThreads, chunkDurationMilli, chunkSizeMb int) float64 {
+	chunkDuration := int64(chunkDurationMilli * milliSecsInSecond)
+	chunkSize := int64(chunkSizeMb * bytesInMB)
+	chunkStatus := ChunkStatus{
+		DurationMillis: chunkDuration,
+		Files: []FileUploadStatusResponse{
+			{
+				Status:           Success,
+				ChecksumDeployed: false,
+				SizeBytes:        chunkSize,
+			},
+		},
+	}
+	timeEstMng.addChunkStatus(chunkStatus, workingThreads, true)
+	return calculateChunkSpeed(workingThreads, chunkSize, chunkDuration)
+}
+
+func TestTransferredSizeInState(t *testing.T) {
+	cleanUp := initStateTest(t)
+	defer cleanUp()
+
+	timeEstMng := newDefaultTimeEstimationManager()
+
+	// Create repos in state.
+	assert.NoError(t, resetRepoState(repo1Key))
+	assert.NoError(t, resetRepoState(repo2Key))
+	saveAndAssertTransferredSizes(t, timeEstMng, 0, 0)
+
+	// Add a chunk of repo1 with multiple successful files, which are included in total.
+	chunkStatus1 := ChunkStatus{
+		DurationMillis: 10 * milliSecsInSecond,
+		Files: []FileUploadStatusResponse{
+			{
+				FileRepresentation: FileRepresentation{
+					Repo: repo1Key,
+				},
+				Status:           Success,
+				ChecksumDeployed: false,
+				SizeBytes:        10 * bytesInMB,
```

Comment:
This initialization repeats several times in this file. It also blows the functions' lengths.
To increase readability, let's use a new function `createFileUploadStatusResponse`.